### PR TITLE
pkg: expose p2p functions

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

Expose p2p functions for use in the sdk. 

These functions could also be copied over to the sdk. I dont have a preference of which is better.

### CODE REVIEW

**Review by alexanderbez — APPROVED** [review]

**Comment by tychoish** [comment]
I think when I looked at this earlier, that the new constructors in `node` actually obviated the need for this to be public. I think it's possible that we could avoid needing to do this.

**Comment by marbar3778** [comment]
> I think when I looked at this earlier, that the new constructors in `node` actually obviated the need for this to be public. I think it's possible that we could avoid needing to do this. 

Could you eloborate. These are mainly helper functions that grab the node key. This isn't possible on the node since it's a service.

**Comment by tychoish** [comment]
Can you link the places where you thought we needed this? I saw a bunches
of places that called into this in the SDK and I thought that it would be
possible to avoid these calls entirely.

File: pkg/p2p/key.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"io/ioutil"
+
+	"github.com/tendermint/tendermint/internal/p2p"
+	tmjson "github.com/tendermint/tendermint/libs/json"
+	tmos "github.com/tendermint/tendermint/libs/os"
+	"github.com/tendermint/tendermint/types"
+)
+
+// LoadNodeKey loads NodeKey located in filePath.
+func LoadNodeKey(filePath string) (p2p.NodeKey, error) {
```
Comment (tychoish) [inline pkg/p2p/key.go:13]: This has a dependency on an internal type (`p2p.NodeKey`) which is internal (and therefore difficult to use by callers outside of this package, and is (confusingly) sort of the same as having a function that has `func Exported() unexported {}`, which is allowable by the compiler but not super useful (and I think shouldn't be allowed, but here we are).

**Review by tychoish — CHANGES_REQUESTED** [review]
I think it would make more sense to: 
- Add a method to the config object that produced the `types.NodeID` from a config object, which means `p2p.LoadNodeKey` no longer needs to get exported. 
- Moved `InitializeNodeValidatorFilesFromMnemonic` function from the SDK to the `config` directory into tendermint. 

I think this is pretty good UX, and avoids the problem of `p2p.NodeKey` needing to also become public.

File: pkg/p2p/key.go (hunk 1 above)
Comment (alexanderbez) [reply pkg/p2p/key.go:13]: Good catch.

**Comment by marbar3778** [comment]
> * Add a method to the config object that produced the `types.NodeID` from a config object, which means `p2p.LoadNodeKey` no longer needs to get exported.

> * Moved `InitializeNodeValidatorFilesFromMnemonic` function from the SDK to the `config` directory into tendermint.

Will look into this. We should be able to extend the config and then modifiy the function in the sdk. No need to move things into tendermint.

File: pkg/p2p/key.go (hunk 1 above)
Comment (marbar3778) [reply pkg/p2p/key.go]: now it works only with nodeID which is public

**Comment by tychoish** [comment]
This is looking better!

> Will look into this. We should be able to extend the config and then modifiy the function in the sdk. No need to move things into tendermint.

I'm excited to see how this goes! It strikes me as possible that the compiler won't allow it, but it strikes me as wrong that this was ever in the `p2p` package anyway. 

I'm not sure that moving the function into tendermint is bad: looking at it, it seems like the kind of thing that any user of tendermint  would end up needing/wanting. I think my first response to this is "this is a pretty small/low-level function to expose publically, and the function in the SDK is all about managing the node rather than interacting with ABCI (which is the core goal of the SDK, as I understand it.)

**Comment by marbar3778** [comment]
> Moved InitializeNodeValidatorFilesFromMnemonic function from the SDK to the config directory into tendermint.

This function has specific stuff to the sdk. We would also need to break the crypto interface to support generating the key from the seed phrase. I dont think its worth messing around with this.

**Comment by tychoish** [comment]
> Also adding the loadkey and loadorgenkey on the config gets a circular dependency with p2p. Not sure if we want to move things around drastically?

I wouldn't characterize anything at this point as drastic, because we're already moving packages around, so I think it's safe to do. 

I'm not sure I see the cylce that you're talking about? those functions have very few dependencies, but maybe something about tests or the switch that I'm missing?

**Review by tychoish — APPROVED** [review]

**Review by marbar3778 — COMMENTED** [review]
changelog my friend

**Pull request merged**